Automattic/woocommerce-payments
# Fix Composite product-type check in `WC_Payments_Payment_Request_Button_Handler ::is_product_supported()`

Fixes #3382 

#### Changes proposed in this Pull Request

In the `WC_Payments_Payment_Request_Button_Handler ::is_product_supported()` method, there is a check to validate if a product is a Composite Product. However, this check uses the `is_composite_product()` that works on the `global $product` object that is not yet defined. Instead, the `$product` object that is set inside `WC_Payments_Payment_Request_Button_Handler ::is_product_supported()` should be used. 

#### Testing instructions

1. Enable the quick-pay buttons from the WooCommerce Payments settings. 
2. Create a new Composite Product.

Before: 
These quick-pay buttons used to show up for Composite Products. 

After: 
These quick-pay buttons should not show up for Composite Products. 

-------------------

- [x] Covered with tests (there weren't any tests for this before)
